src/util/http/websocket/QueryToSocketDistributor.h

```diff
@@ -62,12 +56,14 @@ class QueryToSocketDistributor {
 
   /// Appends specified data to the vector and signals all waiting websockets
   /// that new data is available
-  net::awaitable<void> addQueryStatusUpdate(std::string payload);
+  void addQueryStatusUpdate(std::string payload);
 
   /// Sets the signal that no new updates will be pushed. This causes any
   /// subsequent calls to waitForUpdate to return immediately if all data
   /// has already been consumed
-  net::awaitable<void> signalEnd();
+  void signalEnd();
+
+  auto strand() const { return strand_; }
```

Comment:
A short comment would be nice explaining why the strand is exposed here and what the implications are and why they don't turn out to be a problem